### PR TITLE
Fix issue 19171 - Fix invalid assert in Array!bool slicing

### PR DESCRIPTION
**Pull request by schveiguy**

### CODE REVIEW

File: std/container/array.d [hunk 1]
```diff
@@ -1664,8 +1664,9 @@ if (is(Unqual!T == bool))
         /// ditto
         Range opSlice(size_t low, size_t high)
         {
+            // Note: indexes start at 0, which is equivalent to _a
```
Comment (n8sh) [inline std/container/array.d]: I would suggest rewording this comment as "`// Note: indices are offsets from _a`" or just removing it.

Comment (schveiguy) [reply std/container/array.d]: Sorry, didn't get to this. It is simply an internal comment to try and express the issue that was there before, so probably not worth a followup.
